### PR TITLE
Try not to follow symbolic link for TopLevelAbsPath

### PR DESCRIPTION
**Pull request by anthonyfok**
See gohugoio/hugo#4538

I think this should do the trick to fix snap build issue while keeping TopLevelAbsPath's output format intact.

Tested on Debian GNU/Linux, Windows 10 and macOS High Sierra.

On Linux and macOS:

After fetching Hugo with `go get`, ~/go/src/github.com/gohugoio/hugo is moved to ~/git/hugo and then is then symlinked to it.  Then, from  ~/go/src/github.com/gohugoio/hugo,
`go test -v -run TestPageWithLastmodFromGitInfo ./hugolib` is run.

Before:

<pre>
Anthonys-Mac-mini:hugo foka$ go test -v -run TestPageWithLastmodFromGitInfo ./hugolib
=== RUN   TestPageWithLastmodFromGitInfo
workingDir = /Users/foka/go/src/github.com/gohugoio/hugo/hugolib/testsite
repoPath = /Users/foka/git/hugo
contentDir = Users/foka/go/src/github.com/gohugoio/hugo/hugolib/testsite/content
--- FAIL: TestPageWithLastmodFromGitInfo (1.16s)
	page_test.go:938: 
			Error Trace:	page_test.go:938
			Error:      	Not equal: 
			            	expected: "2018-03-11"
			            	actual  : "2018-02-28"
			Test:       	TestPageWithLastmodFromGitInfo
FAIL
FAIL	github.com/gohugoio/hugo/hugolib	1.281s
</pre>

After:

<pre>
Anthonys-Mac-mini:hugo foka$ go test -v -run TestPageWithLastmodFromGitInfo ./hugolib
=== RUN   TestPageWithLastmodFromGitInfo
workingDir = /Users/foka/go/src/github.com/gohugoio/hugo/hugolib/testsite
repoPath = /Users/foka/go/src/github.com/gohugoio/hugo
contentDir = hugolib/testsite/content
--- PASS: TestPageWithLastmodFromGitInfo (1.13s)
PASS
ok  	github.com/gohugoio/hugo/hugolib	1.255s
</pre>

### CODE REVIEW

**Comment by anthonyfok** [comment]
(@bep, Wow, thank you for merging this so quickly!  But anyway, for completeness sake, I'll continue with posting the result on Windows.)

On Windows 10:

Similarly, after moving the hugo git directory, and creating the directory symlink in CMD with:

    MKLINK /D C:\Users\Anthony\go\src\github.com\gohugoio\hugo C:\Users\Anthony\git\hugo`

`go test -v -run TestPageWithLastmodFromGitInfo ./hugolib` is run from within C:\Users\Anthony\go\src\github.com\gohugoio\hugo:

Before:

<pre>
Anthony@Anthony-V5 MINGW64 ~/go/src/github.com/gohugoio/hugo (master)
$ go test -v -run TestPageWithLastmodFromGitInfo ./hugolib
=== RUN   TestPageWithLastmodFromGitInfo
workingDir = C:\Users\Anthony\git\hugo\hugolib\testsite
repoPath = C:\Users\Anthony\git\hugo
contentDir = hugolib/testsite/content
--- PASS: TestPageWithLastmodFromGitInfo (1.02s)
PASS
ok      _/C_/Users/Anthony/git/hugo/hugolib     3.525s
</pre>

After:

<pre>
Anthony@Anthony-V5 MINGW64 ~/go/src/github.com/gohugoio/hugo (master)
$ go test -v -run TestPageWithLastmodFromGitInfo ./hugolib
=== RUN   TestPageWithLastmodFromGitInfo
workingDir = C:\Users\Anthony\git\hugo\hugolib\testsite
repoPath = C:\Users\Anthony\git\hugo
contentDir = hugolib/testsite/content
--- PASS: TestPageWithLastmodFromGitInfo (1.15s)
PASS
ok      _/C_/Users/Anthony/git/hugo/hugolib     3.527s
</pre>

So, basically, exactly the same as what `git rev-parse --show-toplevel` outputs.  It seems os.Getwd() on Windows would follow the symlink regardless.

Last but not least, credit goes to https://stackoverflow.com/questions/39746533/git-rev-parse-show-toplevel-to-not-resolve-symbolic-links for suggesting the use of `--show-cdup` instead of `--show-toplevel`.

**Comment by bep** [comment]
Thanks for this. We have some similar reports on the Hugo tracker. So this might solve real user issues.
